### PR TITLE
[Merged by Bors] - feat: port/CategoryTheory.PUnit

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2326
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2326
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2326~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/PUnit.lean [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Functor.Const
+import Mathlib.CategoryTheory.DiscreteCategory
+
+/-!
+# The category `Discrete PUnit`
+
+We define `star : C ⥤ Discrete PUnit` sending everything to `punit.star`,
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/PUnit.lean:17]:
```suggestion
We define `star : C ⥤ Discrete PUnit` sending everything to `Punit.star`,
```

**Review by jcommelin — APPROVED** [review]
bors r+

**Pull request closed without merging**